quickwit/quickwit-control-plane/src/indexing_plan.rs

```diff
@@ -192,24 +166,29 @@ pub(crate) fn build_physical_indexing_plan(
         // source.
         let best_node_opt: Option<&str> = candidates
             .iter()
-            // NodeScore implements Ord and can be used easily for sorting.
-            .map(|node_id| NodeScore {
-                node_id,
-                score: -compute_node_score(node_id, &plan),
-            })
-            .sorted()
+            .map(|&node_id| (node_id, compute_node_score(node_id, &plan)))
+            // Note(fmassot): we can't use max_by as, in case of equality, it returns the last
+            // element and we want to preserve the order given by the rendez-vous hashing.
+            .sorted_by(cmp_node_by_reverse_score)
```

Comment:
```suggestion
            .rev() //< we use the reverse iterator, because in case of a tie, max picks the first element.
            .map(|&node_id| (node_id, compute_node_score(node_id, &plan)))
            .max_by_key(cmp_node_by_reverse_score)
```